### ISSUE TITLE
[<Ray component: Data>] Unable to convert iter_batches to TensorFlow from parquet dataset when it contain list object.

### ISSUE THREAD

**Issue by MoFHeka** [ray-project/ray#48316]
### What happened + What you expected to happen

If we use ray.data.read_parquet to read a column which object is a list, and iterate it with batches. Then it will return batch size number of Numpy arrays in one Numpy array.
When an array was read batches from parquet file, it shouldn't be a array([array()], object) which can't not be convert to TensorFlow tensor. But a NDArray with correct dimension [batch_size, dim].
ray.data.Dataset.to_tf failed, and no way to use Ray data to train model.

### Versions / Dependencies

ray==2.38.0
tensorflow==2.15

### Reproduction script

'''python
import pandas as pd
tmp_path = '/tmp/4272932fdf4bc1be466053414a71e0ae/'
df1 = pd.DataFrame({"one": [[1,1], [2,2], [3,3]], "two": ["a", "b", "c"]})
table = pa.Table.from_pandas(df1)
path1 = os.path.join(str(tmp_path), "test1.parquet")
pq.write_table(table, path1)
ds = ray.data.read_parquet([path1])
for batch in ds.iter_batches(batch_size=2):
    print(batch) # Here we see that 'one' is a array([array()])
    try:
        tf.convert_to_tensor(batch['one'])
    except Exception as e:
        print(e) # Failed to convert a NumPy array to a Tensor (Unsupported object type numpy.ndarray).
'''

### Issue Severity

High: It blocks me from completing my task.

**Comment by wingkitlee0** [comment]
Not sure if this is Ray related. You will get a similar result with a pyarrow Table and `list`-type column (e.g., the `table` in your example).

try ` tf.convert_to_tensor(np.vstack(batch["one"].to_numpy()))`?
